### DISCUSSION TITLE
How can i rotate this image with an icon ?

### DISCUSSION THREAD

**Discussion by BennyTian** [software-mansion/react-native-gesture-handler#2529] [Q&A] [answered]
How can i rotate this image with an icon ?

![3651688541375_ pic](https://github.com/software-mansion/react-native-gesture-handler/assets/3095848/f5cb07c3-7c26-4062-91e0-349f9d2256db)

**Comment by danvernon** [comment]
@BennyTian did you figure this out?

**Comment by xiaoosnggao** [comment]
@BennyTian  The same problem

**Comment by Glazzes** [comment]
It can be achieved with some simple trigonometry and a pan gesture in the following manner:
- Detertmine the angle formed by the rotation icon and the center of the image
- Make sure the rotation icon has a pan gesture attached to it
- Now add the translation values to the position of the rotation icon, from there you can calculate the angle based on the new position
- With both angles get the difference from inital angle and the current one, then rotate the image by the difference between these two

It's hard to explain, but this how one achieves rotation with a single finger.

**Reply by xiaoosnggao** [reply]
Excuse me, do you have a demo for this?

**Reply by Glazzes** [reply]
@xiaoosnggao  I can not recall where I placed that file as it was never uploaded, however I found a video I recorded and sent to my partner of that demo, If I manage to find it I will come back to you.

This is the demo video in question, I recall I used the techniques I mentioned.

https://github.com/user-attachments/assets/dddd154d-393a-4cf4-9592-6e667e28b351

**Reply by xiaoosnggao** [reply]
@Glazzes Thank you very much

**Reply by Glazzes** [reply, marked as answer]
@xiaoosnggao I managed to find the file and it was a mess, I took to the time to rewrite it for a simple demo of my own. This is the most complete version I can share as a stand alone file https://gist.github.com/Glazzes/3249c28b8c33e300df0a53273db7dc78

**Reply by risardi-lahelu** [reply]
Thanks @Glazzes, really appreciate!
